### ISSUE TITLE
forked-daapd crashing / not working for some internet radio streams

### ISSUE THREAD

**Issue by snizzleorg** [owntone/owntone-server#62]
I frequently have problems with the following stream:
http://stream.dradio.de/7/249/142684/v1/gnl.akacast.akamaistream.net/dradio_mp3_dlf_m

while others like:
http://broadcast.infomaniak.net/radionova-high.mp3
work every single time.

After the playback failed the speakers are all switched off, regardless of which was switched on before the failed play attempt

Does someone have the same problem?

**Comment by chme** [comment]
Today I ran into this problem too, what reduces the time to open the stream is setting the probesize in transcode.c on the AVFormatContext given to avformat_open_input:

[code]
  ctx->fmtctx = avformat_alloc_context();
  ctx->fmtctx->probesize = 10000;
  ctx->fmtctx->max_analyze_duration = 32;
[/code]

Not sure what values are reasonable or if you want to change them from the defaults ...

**Comment by ejurgensen** [comment]
I don't mind changing the defaults for versions of ffmpeg/libav which have the problem (would require some testing to find out where to set the cutoff libavformat version). If I understand the docs then max_analyse_duration is in microseconds, and its default is 5000000? And I understand probesize to be in bytes and with no default?

**Comment by chme** [comment]
For testing different values: 
https://github.com/chme/forked-daapd/tree/probesize

Values can be configured in `forked-daapd.conf` by adding:
```
avformat {
	probesize = 10000
	max_analyze_duration = 32
}
```

**Comment by chme** [comment]
According to https://patches.libav.org/patch/32429/ the default for probesize is 5000000 bytes. 
Minimum values are 32 for both. As the information of the stream duration is not used, setting it to 32 should be ok? On my ubuntu laptop a probesize of 10000 worked fine. Will do more tests on my rpi.

**Comment by ejurgensen** [comment]
Thanks for the testing branch.

Did a bit of testing (libav 9.14 and the stream above + the streams provided by the RPi user), and for me max_analyze_duration doesn't do anything, while reducing probesize did wonders. A probesize of 128000 seems equal up to ~4 seconds, while 64000 was about 1 sec, so I'm inclined to go for that. It always detected format, sample rate and so on just fine.

Does max_analyze_duration make a difference for you?

I'll look into making the change in the m3u filescanner as well. For transcode.c I'm not sure if it should be for internet streams only, or for everything.

Now I'll test different versions of libav and ffmpeg.

**Comment by ejurgensen** [comment]
My tests showed that all versions of libav were slow, while all versions of ffmpeg were quick. However, I had trouble compiling ffmpeg < 1.1, so I didn't get to test those.

I did see one misdetection with probesize 64000 (detected format mpl instead of mp3), and going down below 10k would make icy metadata sketchy.

So now I've added a fix, which should apply to just libav. Due to the risk of misdetection I decided to only include internet streams for now.
